js/adapt-contrib-resourcesView.js

```diff
@@ -49,7 +49,20 @@ define(function(require) {
         },
 
         onResourceClicked: function(event) {
-            window.top.open($(event.currentTarget).data("href"));
+            var data = $(event.currentTarget).data();
+
+            if ('document' !== data.type) {
+              window.top.open($(event.currentTarget).data("href"), 'MyFile.jpg');
+              return;
+            }
+            var dummyLink = document.createElement('a');
+            dummyLink.download = data.filename;
```

Comment:
As `filename` is a new property, it will not exist for older courses, so some defensive code is required here.